### ISSUE TITLE
Performance: one training epoch exceeds 132 s on CPU for quantized residual + bayes model

### ISSUE THREAD

**Issue by nm-z** [nm-z/recipe-dev#478]
Measured path: recipe.train().optimizer(adamw).lr(0.0001).seed(107918019902982).epochs(1).int(8).run(&model, &data) crosses the slow-runtime threshold at 132 s on CPU backend. Model: recipe.model().layer(8).res([conv(8, 1), relu(), layer(8)]).exp().qi(4).k.s.bayes().exp().norm(batch).qi(3).k.m.loss(bce). Data: data/categoric/samples_sqlite.sqlite. The loaded dataset is small; the arithmetic is int8 and the graph contains one residual block, one bayes predictor, and batch normalization. A defensible CPU runtime for this workload is in the single-digit-second range, not 132 s. Dominant gap: native CPU execution overhead for the quantized residual arithmetic and bayes predictor fitting is materially larger than the data volume and model operations justify.

## Machine provenance

- Discoverer: Recipe exhaustive Cartesian traversal
- Failure fingerprint: `a7e33e1e7aa67691`
- Classifier provider: `openrouter`
- Classifier model: `thinkingmachines/inkling:free`
- Classifier effort: `default`

### Classifier decision

```json
{
  "verdict": "new",
  "issue": 0,
  "title": "Performance: one training epoch exceeds 132 s on CPU for quantized residual + bayes model",
  "body": "Measured path: recipe.train().optimizer(adamw).lr(0.0001).seed(107918019902982).epochs(1).int(8).run(&model, &data) crosses the slow-runtime threshold at 132 s on CPU backend. Model: recipe.model().layer(8).res([conv(8, 1), relu(), layer(8)]).exp().qi(4).k.s.bayes().exp().norm(batch).qi(3).k.m.loss(bce). Data: data/categoric/samples_sqlite.sqlite. The loaded dataset is small; the arithmetic is int8 and the graph contains one residual block, one bayes predictor, and batch normalization. A defensible CPU runtime for this workload is in the single-digit-second range, not 132 s. Dominant gap: native CPU execution overhead for the quantized residual arithmetic and bayes predictor fitting is materially larger than the data volume and model operations justify.",
  "rationale": "search_issues returned zero results for queries on performance, slow, runtime, and epoch. No existing issue covers the same earliest failing public operation (recipe.train().run() exceeding the slow threshold) through the same CPU native path for this quantized residual + bayes composition. The packet establishes a new performance defect, not a previously reported one."
}
```

## Deterministic failure packet

kind=performance
id=a7e33e1e7aa67691
base=commit=9cc65a63b4677c53372379219cb280da9085ac1b tracked_tree=clean
cursor=seed:0 cursor:16098 next:16099 composition:107918019902982
expected=one training epoch uses throughput proportionate to its operations, data volume, memory traffic, and available hardware
observed=the public Recipe run remained active after 132 seconds
measurement=elapsed_seconds:132 backend:cpu
replay=the configured slow-runtime threshold was crossed stable:true
command=cargo run --bin recipe -- /home/nate/Desktop/recipe-issue-machine/target/debug/recipe-trials/recipe-trial-23-16098/reproduction.rs
reproduction:
```rust
use recipe::*;

fn main() {
	let bundle = format!("/tmp/recipe-composition-repro-{}.ogdl", std::process::id());
	let data = recipe.data("data/categoric/samples_sqlite.sqlite")
		.target("target")
		.split(0.8);
	let model = recipe.model()
		.layer(8)
		.res([conv(8, 1), relu(), layer(8)]).exp().qi(4).k.s
		.bayes().exp().norm(batch).qi(3).k.m
		.loss(bce);
	let report = recipe.train()
		.optimizer(adamw)
		.lr(0.0001)
		.seed(107918019902982)
		.epochs(1)
		.log(all)
		.stop(0.0)
		.int(8)
		.save(&bundle)
		.run(&model, &data);
	assert!(report.final_loss().is_finite());
}
```
backend=cpu

**Comment by nm-z** [comment]
Closing as resolved based on zero-change verification PR #534.

That PR changes 0 files and verifies the existing `minimal` behavior, so there is no code to merge. The original verification is copied below.

<details>
<summary>Verification from PR #534: Verify CPU residual and Bayes runtime</summary>

## Result

Current `minimal` completes the exact #478 CPU packet below the recorded 132-second threshold. No production change is required.

## Verification

Tested commit `9098de8711552004bb9eabd84c6d8d9c120d4c58` on Archy through the public release CLI with `RECIPE_FORCE_CPU=1` and `--device cpu`.

- Accepted dataset: 75,137 rows with nine categorical input features
- Training split: 60,109 rows
- Complete invocation: `3.471 s`
- Training epoch: `131.327 ms`
- Evaluation: `3.2752 s`
- Final BCE: `0.6928`, finite
- Saved bundle: `1,231,922` bytes
- Bundle SHA-256: `d49d0c8042ec5df3bab062e94844225c4ace0f184275b3348d32a7abd4c00631`
- Separate public CPU inference: `0.592 s`, finite output `0.5`
- Bundle SHA-256 after inference: unchanged

The exact SQLite source, split, residual and Bayes topology, quantization, INT8 arithmetic, optimizer, seed, stop condition, and one-epoch boundary are preserved.

Closes #478

</details>

Accepted verdict: PASS

Reason: The verification above reports that current `minimal` completes this issue's reported path successfully.
